seumich/views.py

```diff
@@ -95,19 +95,27 @@ class AdvisorView(LoginRequiredMixin, UserLogPageViewMixin, ListView):
 
     def get_context_data(self, **kwargs):
         context = super(AdvisorView, self).get_context_data(**kwargs)
-        context['studentListHeader'] = self.mentor.first_name + \
-            " " + self.mentor.last_name
-        context['advisor'] = self.mentor
+        user = User.objects.get(username=self.kwargs['advisor'])
+        if self.mentor is not None:
+            context['studentListHeader'] = " ".join([self.mentor.first_name, self.mentor.last_name])
+            context['advisor'] = self.mentor
+        else:
+            context['studentListHeader'] = " ".join([user.first_name, user.last_name])
+            context['advisor'] = user
         return context
 
     def get_queryset(self):
-        self.mentor = get_object_or_404(Mentor,
-                                        username=self.kwargs['advisor'])
-        student_list = self.mentor.students.order_by('last_name').distinct()
-        student_list = student_list.prefetch_related(
-            'studentclasssitestatus_set__status',
-            'studentclasssitestatus_set__class_site',
-            'cohorts')
+        self.mentor = None
+        try:
+            self.mentor = Mentor.objects.get(username=self.kwargs['advisor'])
+            student_list = self.mentor.students.order_by('last_name').distinct()
```

Comment:
```suggestion
            student_list = self.mentor.students.filter(id>0).order_by('last_name').distinct()
```
May as well implement this now to get rid of "Not Happened Yet" as a student in the list

Reply:
Okay, I can do this, though it is a separate issue. How are you planning on identifying the instances of Not Happened Yet in the future? It's nice to not have them show up, but then we don't know that something went wrong and that an advisor doesn't have a student they want in their cohort. I'm guessing you have a separate way of determining this?

Reply:
It is ok to leave out for now. It's not urgent or anything. I just figured it was such a small change to code we're already changing. I don't mind if you reject.

The utility of the appearance of "Not Happened Yet" is to make it very apparent that "something is wrong" which can cue me to run that program to figure out which uniqname(s) caused the error. However, I discovered it's much easier to filter BG_STDNT_CHRT_MNTR by STDNT_KEY<1 which will tell me which cohorts I need to look at.
![image](https://user-images.githubusercontent.com/31591389/66077806-b89c0600-e52e-11e9-9f35-a0f4073545e8.png)

Reply:
I think you're right. It's a small change, and you have other ways to check for those values. I'm working on this now; I also want to check that there aren't other places with student QuerySets that need to be filtered.

Reply:
Done. See [this commit](https://github.com/tl-its-umich-edu/student_explorer/pull/161/commits/dfd11a2cd8c18310e69961fbbce71ad15e4684d8). We'll need to review this carefully in test, since there aren't zero or negative values for `Student.id` in the default data in my local database.